lib/lightning/collections.ex

```diff
@@ -209,16 +208,57 @@ defmodule Lightning.Collections do
     end
   end
 
+  @spec delete_all(Collection.t(), String.t()) ::
+          {:ok, non_neg_integer()} | {:error, :not_found}
+  def delete_all(%{id: collection_id}, key_pattern \\ nil) do
+    query =
+      from(i in Item, where: i.collection_id == ^collection_id)
+      |> then(fn query ->
+        case key_pattern do
+          nil -> query
+          pattern -> where(query, [i], like(i.key, ^format_pattern(pattern)))
+        end
+      end)
+
+    case Repo.delete_all(query) do
+      {0, nil} -> {:error, :not_found}
+      {n, nil} -> {:ok, n}
+    end
+  end
+
   defp stream_query(collection_id, cursor, limit) do
     Item
     |> where([i], i.collection_id == ^collection_id)
-    |> order_by([i], asc: i.updated_at)
+    |> order_by([i], asc: i.inserted_at)
     |> limit(^limit)
     |> then(fn query ->
       case cursor do
         nil -> query
-        ts_cursor -> where(query, [i], i.updated_at > ^ts_cursor)
+        ts_cursor -> where(query, [i], i.inserted_at > ^ts_cursor)
       end
     end)
   end
+
+  defp filter_by_inserted_at(query, params) do
+    query
+    |> filter_by_created_before(params)
+    |> filter_by_created_after(params)
+  end
+
+  defp filter_by_created_after(query, %{created_after: created_after}),
+    do: where(query, [i], i.inserted_at >= ^created_after)
+
+  defp filter_by_created_after(query, _params), do: query
+
+  defp filter_by_created_before(query, %{created_before: created_before}),
+    do: where(query, [i], i.inserted_at < ^created_before)
+
+  defp filter_by_created_before(query, _params), do: query
```

Comment:
Would this benefit from using `Ecto.Query.dynamic/2`?

Reply:
For this single known condition I've followed the approach to keep as simple as possible while making it modular and maintainable. But if we gonna have more complex queries in the future we could make a dynamic reduce in advance.